src/utils/resources/template/hooks/useVmTemplateSource/useVmTemplateSource.ts

```diff
@@ -81,7 +83,9 @@ export const useVmTemplateSource = (template: V1Template): useVmTemplateSourceVa
       .finally(() => setLoaded(true));
   };
 
-  React.useEffect(() => {
+  useEffect(() => {
+    if (isEqualObject(prevBootSourceRef?.current, bootSource)) return;
```

Comment:
The useEffect (also without the sidebar editor) run multiple time as the template change during rendering. With sidebar editor, this would run at every keystroke. So run only if the boot source actually changes. 
Don't think `isEqualObject` is heavy. The objects are small